nni/algorithms/compression/v2/pytorch/pruning/auto_compress_pruner.py

```diff
@@ -0,0 +1,112 @@
+    """
+    Parameters
+    ----------
+    model : Module
+        The origin unwrapped pytorch model to be pruned.
+    config_list : List[Dict]
+        The origin config list provided by the user. Note that this config_list is directly config the origin model.
+        This means the sparsity provided by the origin_masks should also be recorded in the origin_config_list.
+    total_iteration : int
+        The total iteration number.
+    evaluator : Callable[[Module], float]
+        Evaluate the pruned model and give a score.
+    admm_params : Dict
+        The parameters pass to the ADMMPruner.
+
+        - trainer : Callable[[Module, Optimizer, Callable].
+            A callable function used to train model or just inference. Take model, optimizer, criterion as input.
+            The model will be trained or inferenced `training_epochs` epochs.
+        - optimizer : torch.optim.Optimizer.
+            The optimizer instance used in trainer. Note that this optimizer might be patched during collect data,
+            so do not use this optimizer in other places.
+        - criterion : Callable[[Tensor, Tensor], Tensor].
+            The criterion function used in trainer. Take model output and target value as input, and return the loss.
+        - iterations : int.
+            The total iteration number in admm pruning algorithm.
+        - training_epochs : int.
+            The epoch number for training model in each iteration.
+
+    sa_params : Dict
+        The parameters pass to the SimulatedAnnealingPruner.
+
+        - evaluator : Callable[[Module], float]. Required.
+            Evaluate the pruned model and give a score.
+        - start_temperature : float. Default: `100`.
+            Start temperature of the simulated annealing process.
+        - stop_temperature : float. Default: `20`.
+            Stop temperature of the simulated annealing process.
+        - cool_down_rate : float. Default: `0.9`.
+            Cool down rate of the temperature.
+        - perturbation_magnitude : float. Default: `0.35`.
+            Initial perturbation magnitude to the sparsities. The magnitude decreases with current temperature.
+        - pruning_algorithm : str. Default: `'level'`.
+            Supported pruning algorithm ['level', 'l1', 'l2', 'fpgm', 'slim', 'apoz', 'mean_activation', 'taylorfo', 'admm'].
+        - pruning_params : Dict. Default: `{}`.
+            If the pruner corresponding to the chosen pruning_algorithm has extra parameters, put them as a dict to pass in.
+
+    log_dir : str
+        The log directory use to saving the result, you can find the best result under this folder.
+    keep_intermediate_result : bool
+        If keeping the intermediate result, including intermediate model and masks during each iteration.
+    finetuner : Optional[Callable[[Module], None]]
+        The finetuner handled all finetune logic, use a pytorch module as input, will be called in each iteration.
```

Comment:
handled -> handles, use -> take

Reply:
", will be ..." -> ". It will be called at the end of each iteration, usually for neutralizing the accuracy loss brought by the pruning in this iteration."

Reply:
fix it